omnipaxos_core/src/util.rs

```diff
@@ -88,6 +91,27 @@ where
         (pid - 1) as usize
     }
 
+    // Resets `pid`'s accept sequence to indicate they are in prepare phase
+    pub fn reset_accept_sequence(&mut self, pid: NodeId) {
+        self.accept_sequences[Self::pid_to_idx(pid)] = 0;
+    }
+
+    pub fn get_acceptsync_sequence_num(&mut self, pid: NodeId) -> u64 {
+        if self.accept_sequences[Self::pid_to_idx(pid)] != 0 {
+            panic!("AcceptSync must be first message in accept sequence");
+        }
+        self.accept_sequences[Self::pid_to_idx(pid)] = 1;
+        1
+    }
```

Comment:
This function is confusing since it is called `get_`... but actually also sets the sequence number of the node on line 103. I also think the function is unnecessary, and it would be more clean and clear to just use a constant for the AcceptSync sequence number (and set/reset) where needed.

`const ACCSYNC_SEQNUM: u64 = 1;`

```suggestion
```

Reply:
I think the LeaderState API just having `reset_seq_num` and `next_seq_num` could be clean. Then when sending accepts you just ask for the next seq number and when sending AcceptSyncs you reset the sequence and then ask for the next seq number. It becomes less clear that AcceptSync always has seq 1, but you don't have to define a third function `set_seq_num` and don't have users of LeaderState need to know what value to set the sequence to.